modules/bmp/image_loader_bmp.cpp

```diff
@@ -129,35 +149,27 @@ Error ImageLoaderBMP::load_image(Ref<Image> p_image, FileAccess *f,
 			bmp_header.bmp_info_header.bmp_colors_used = f->get_32();
 			bmp_header.bmp_info_header.bmp_important_colors = f->get_32();
 
-			bmp_header.bmp_info_header.bmp_red_mask = f->get_32();
-			bmp_header.bmp_info_header.bmp_green_mask = f->get_32();
-			bmp_header.bmp_info_header.bmp_blue_mask = f->get_32();
-			bmp_header.bmp_info_header.bmp_alpha_mask = f->get_32();
-			bmp_header.bmp_info_header.bmp_cs_type = f->get_32();
-			for (int i = 0; i < 9; i++)
-				bmp_header.bmp_info_header.bmp_endpoints[i] = f->get_32();
-
-			bmp_header.bmp_info_header.bmp_gamma_red = f->get_32();
-			bmp_header.bmp_info_header.bmp_gamma_green = f->get_32();
-			bmp_header.bmp_info_header.bmp_gamma_blue = f->get_32();
+			// Compressed bitmaps not supported, stop parsing
+			if (bmp_header.bmp_info_header.bmp_compression != BI_RGB) {
+				ERR_EXPLAIN("Unsupported bmp file: " + f->get_path());
+				f->close();
+				ERR_FAIL_V(err)
+			}
 
-			f->seek(sizeof(bmp_header.bmp_file_header) +
+			f->seek(FILE_HEADER_SIZE +
 					bmp_header.bmp_info_header.bmp_header_size);
 
-			uint32_t color_table_size = 0;
-			if (bmp_header.bmp_info_header.bmp_bit_count == 1)
-				color_table_size = 2;
-			else if (bmp_header.bmp_info_header.bmp_bit_count == 4)
-				color_table_size = 16;
-			else if (bmp_header.bmp_info_header.bmp_bit_count == 8)
-				color_table_size = 256;
+			if (bmp_header.bmp_info_header.bmp_bit_count < 16 && bmp_header.bmp_info_header.bmp_colors_used == 0)
+				bmp_header.bmp_info_header.bmp_colors_used = 1 << bmp_header.bmp_info_header.bmp_bit_count;
```

Comment:
I don't think `bmp_header.bmp_info_header.bmp_colors_used` should be overwritten that way, and better be explicit about color table size because some malicious bmp's may override `bmp_bit_count` to overflow max buffer size I think.